backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2NativeCompilerArguments.kt

```diff
@@ -126,6 +126,9 @@ class K2NativeCompilerArguments : CommonCompilerArguments() {
     )
     var exportedLibraries: Array<String>? = null
 
+    @Argument(value = "-Xlight-debug", description = "Add light debug information")
```

Comment:
Not very descriptive name, maybe `-g0` or `-Xrelease-backtrace`?

Reply:
I think it's better to enable this behaviour by default and add option to omit debug info instead.

Reply:
I agree, especially since later build stages can run `strip` to remove all of it.

Reply:
> Not very descriptive name, maybe -g0 or -Xrelease-backtrace?

It's a draft, the name is not chosen yet.

> I think it's better to enable this behaviour by default and add option to omit debug info instead.

1. I believe it makes sense to release it first as experimental (and thus optional) to ensure it doesn't break anything.
2. Consider the effect of `-Xlight-debug` on non-Apple platforms.

Reply:
Even on non-Apple platforms, debug info can be stripped back out (even automatically at a later stage: generating the k/n binary), no? Seems like risk of making this opt-in is greater: every library has to opt-in to ensure complete debug info in production.

Reply:
> Even on non-Apple platforms, debug info can be stripped back out (even automatically at a later stage: generating the k/n binary)

I mean current implementation. Consider the effect of enabling this option on non-Apple platforms using current implementation.

> every library has to opt-in to ensure complete debug info in production

Note that there is no debug info in libraries currently, it is added at "a later stage: generating the k/n binary".

Reply:
I see. I think I understand now. 👍